auto_submit/test/requests/github_webhook_test_data.dart

```diff
@@ -303,6 +303,2303 @@ String webhookEventMock = '''{
+              "has_pages": false,
+              "has_downloads": true,
+              "archived": false,
+              "disabled": false,
+              "pushed_at": "2011-01-26T19:06:43Z",
+              "created_at": "2011-01-26T19:01:12Z",
+              "updated_at": "2011-01-26T19:14:43Z",
+              "permissions": {
+                "admin": false,
+                "push": false,
+                "pull": true
+              },
+              "allow_rebase_merge": true,
+              "temp_clone_token": "ABTLWHOULUVAXGTRYU7OC2876QJ2O",
+              "allow_squash_merge": true,
+              "allow_merge_commit": true,
+              "forks": 123,
+              "open_issues": 123,
+              "license": {
+                "key": "mit",
+                "name": "MIT License",
+                "url": "https://api.github.com/licenses/mit",
+                "spdx_id": "MIT",
+                "node_id": "MDc6TGljZW5zZW1pdA=="
+              },
+              "watchers": 123
+            }
+          },
+          "_links": {
+            "self": {
+              "href": "https://api.github.com/repos/octocat/Hello-World/pulls/1347"
+            },
+            "html": {
+              "href": "https://github.com/octocat/Hello-World/pull/1347"
+            },
+            "issue": {
+              "href": "https://api.github.com/repos/octocat/Hello-World/issues/1347"
+            },
+            "comments": {
+              "href": "https://api.github.com/repos/octocat/Hello-World/issues/1347/comments"
+            },
+            "review_comments": {
+              "href": "https://api.github.com/repos/octocat/Hello-World/pulls/1347/comments"
+            },
+            "review_comment": {
+              "href": "https://api.github.com/repos/octocat/Hello-World/pulls/comments{/number}"
+            },
+            "commits": {
+              "href": "https://api.github.com/repos/octocat/Hello-World/pulls/1347/commits"
+            },
+            "statuses": {
+              "href": "https://api.github.com/repos/octocat/Hello-World/statuses/6dcb09b5b57875f334f61aebed695e2e4193db5e"
+            }
+          },
+          "author_association": "OWNER",
+          "auto_merge": null,
+          "draft": false,
+          "merged": false,
+          "mergeable": true,
+          "rebaseable": true,
+          "mergeable_state": "clean",
+          "comments": 10,
+          "review_comments": 0,
+          "maintainer_can_modify": true,
+          "commits": 3,
+          "additions": 100,
+          "deletions": 3,
+          "changed_files": 5
+      },
+      "repository": {
+        "id": 1868532,
+        "name": "flutter",
+        "full_name": "flutter/cocoon",
+        "private": false,
+        "owner": {
+          "login": "flutter",
+          "id": 21031067
+        },
+        "html_url": "https://github.com/cocooon/octocat",
+        "watchers": 0
+      }
+    }''';
+
+final String webhookNoStatusRepoMock = '''{
```

Comment:
Instead of using `.fromJson`, PullRequest can simply use the default constructor to inject only what we need.
Please check if other classes (other than CheckRun) can follow this.